### PR TITLE
DOP-1604: Allow caption to be an option for code blocks

### PR DESCRIPTION
**Pull request by rayangler**
[[DOP-1604](https://jira.mongodb.org/browse/DOP-1604)]

This will allow `:caption:` to be a valid option for code blocks. For now, we can keep captions as strings, but we will later need to be able to support captions with inline code such as in the case of:
```
.. code-block: javascript
   :caption: Test ``Caption``

   foo
```

### CODE REVIEW

File: snooty/gizaparser/release.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 @dataclass
 class ReleaseSpecification(Inheritable):
     pre: Optional[str]
+    caption: Optional[str]
```
[the remark this answers is not in the document]
Comment (rayangler) [reply snooty/gizaparser/release.py:17]: I remember running into errors regarding `release.py` and `steps.py` without caption being added. I can double-check if adding this line is necessary

File: snooty.idl [hunk 2]
```diff
@@ -117,6 +117,7 @@ interface HTML <: Literal {
 interface Code <: Literal {
   type: "code";
   lang: string?;
+  caption: string;
```
[the remark this answers is not in the document]
Comment (rayangler) [reply snooty.idl:120]: Hahaha, would you like me to remove this file in this PR in that case?

File: snooty/gizaparser/release.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (i80and) [reply snooty/gizaparser/release.py:17]: You'll need to pass `None` in for the caption lower in the file

File: snooty.idl (hunk 2 above)
[the remark this answers is not in the document]
Comment (i80and) [reply snooty.idl:120]: Go for it! 🚀

**Review by i80and — APPROVED** [review]
metal 🤘🏻

**Pull request merged**